### DISCUSSION TITLE
Special considerations for transferring to a new server?

### DISCUSSION THREAD

**Discussion by gddgddg** [owncast/owncast#2894] [Q&A] [answered]
I started streaming with an Owncast instance a while ago on an underpowered VPS, but I recently obtained a new server and want to stream there for the benefit of live reencoding. But I have a small following and want to make sure everything goes properly.

My thought is I have to:

  - Transfer the Owncast directory to the new server
  - Point the domain to the new server
  - Set up the reverse proxy and SSL to match what it was on the old server

I haven't tried this yet, but I want to ask here to make sure I don't have to start all over. Please let me know if there's anything I'm missing.

To further specify, I installed Owncast using the quick installer and have it running as a systemd service. I'm not much of a server admin, but I like hosting my own services like this.

**Comment by gabek** [comment, marked as answer]
You should be able to backup your `data` directory and restore it on your new server. I suggest you backup your `data` directory regularly in case of needing to restore it anyway, so this would be no different than that.

**Comment by SpiderD555** [comment]
@gabek I have very similar remark, but related to the Docker-type deployment.

Copy - Paste of the whole data directory works, but when I tried to follow the Backup - Restore instructions from within the docker container, then I got this:
```
/app # ./owncast --restoreDatabase /app/data/backup/owncastdb.bak
INFO[2024-09-25T12:44:49Z] Owncast v0.1.3-linux-64bit (6ae3167b61d8485d3f269bd422f0c1a1f072ea5c) 
INFO[2024-09-25T12:44:49Z] Restoring database backup /app/data/backup/owncastdb.bak to data/owncast.db 
FATA[2024-09-25T12:44:54Z] database is locked
```
I assume that the database lock can be lifted by stopping the container, however that means I won't be able to go into the container to execute _owncast_ command. A chicken and egg problem I think.

**Reply by gabek** [reply]
Oh, if you're just moving over the entire `data` directory then you don't need to restore backups. Unless for some reason the old database won't work on your new server (due to architecture reasons, etc).

You can launch your container with a new entrypoint to gain access to it with a shell, that's probably the best solution if you want gain access to it, but not run your original entrypoint.
